File: Archiver/CS/LoginForm1.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Drawing;
using System.Linq;
using System.Diagnostics;
using System.Data;
using Microsoft.VisualBasic;
using MODI;
using System.Xml.Linq;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using Microsoft.VisualBasic.CompilerServices;


namespace EcmArchiveClcSetup
{
	public partial class LoginForm1
	{
		public LoginForm1()
		{
			// VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
			CurrSessionGuid = Guid.NewGuid();
			
			InitializeComponent();
			
			//Added to support default instance behavour in C#
			if (defaultInstance == null)
				defaultInstance = this;
		}
		
#region Default Instance
		
		private static LoginForm1 defaultInstance;
		
		/// <summary>
		/// Added by the VB.Net to C# Converter to support default instance behavour in C#
		/// </summary>
		public static LoginForm1 Default
		{
			get
			{
				if (defaultInstance == null)
				{
					defaultInstance = new LoginForm1();
					defaultInstance.FormClosed += new FormClosedEventHandler(defaultInstance_FormClosed);
				}
				
				return defaultInstance;
			}
		}
		
		static void defaultInstance_FormClosed(object sender, FormClosedEventArgs e)
		{
			defaultInstance = null;
		}
		
#endregion
		
		// TODO: Insert code to perform custom authentication using the provided username and password
		// (See http://go.microsoft.com/fwlink/?LinkId=35339).
		// The custom principal can then be attached to the current thread's principal as follows:
		//     My.User.CurrentPrincipal = CustomPrincipal
		// where CustomPrincipal is the IPrincipal implementation used to perform authentication.
		// Subsequently, My.User will return identity information encapsulated in the CustomPrincipal object
		// such as the username, display name, etc.
		
		int AutoLoginSecs = 10;
		int CurrLoginSecs = 0;
		clsLogging LOG = new clsLogging();
		clsIsolatedStorage ISO = new clsIsolatedStorage();
		clsEncryptV2 ENC2 = new clsEncryptV2();
		clsDatabase DB = new clsDatabase();
		clsEncrypt ENC = new clsEncrypt();
		clsDma DMA = new clsDma();
		bool bHelpLoaded = false;
		bool bAttached = false;
		bool bPopulateComboBusy = false;
		
		public bool bGoodLogin = false;
		public string UID = "";
		
		Guid CurrSessionGuid; // VBConversions Note: Initial value of "Guid.NewGuid()" cannot be assigned here since it is non-static.  Assignment has been moved to the class constructors.
		
		SVCSearch.Service1Client ProxySRCH = new SVCSearch.Service1Client();
		
		int SecureID = -1;
		
		System.Collections.Generic.List<string> ListOfRepoIS = new System.Collections.Generic.List<string>();
		string strRepos = "";
		
		string FQN = "";
		string CurrUserName = "";
		
		public void LoginForm1_Load(System.Object sender, System.EventArgs e)
		{
			
			this.TopMost = true;
			
			//Check to see if an AUTO-LOGIN file exists.
			
			
			CurrUserName = Environment.UserName;
			
			getPersitData();
			
			if (txtCompanyID.Text.Trim().Length > 0 && cbRepoID.Text.Trim().Length > 0 && pwEncryptPW.Text.Trim().Length > 0)
			{
				btnAttach_Click(null, null);
			}
			
			lblAttachedMachineName.Text = LOG.getEnvVarMachineName();
			lblNetworkID.Text = LOG.getEnvVarNetworkID();
			
			if (txtCompanyID.Text.Trim().Length > 0 && cbRepoID.Text.Trim().Length > 0 && pwEncryptPW.Text.Trim().Length > 0 && txtLoginID.Text.Trim().Trim().Length > 0 && PasswordTextBox.Text.Trim().Trim().Length > 0)
			{
				Timer2.Enabled = true;
			}
			else
			{
				Timer2.Enabled = false;
			}
			
			modGlobals.bRunnner = false;
		}
		
		public void OK_Click(System.Object sender, System.EventArgs e)
		{
			
			if (lblAttached.Text.ToLower().Equals("not attached"))
			{
				VerifySecureDbConnection();
				MessageBox.Show("The Gateway is not Attached, please attach to the Secure Gateway.");
				return;
			}
			
			if (PasswordTextBox.Text.ToUpper().Equals("PASSWORD"))
			{
				if (txtLoginID.Text.Length == 0)
				{
					MessageBox.Show("Please enter your USERID into the Login ID field.");
					return;
				}
				frmPasswordChange.Default.ShowDialog();
				SB.Text = "Please login using your new credentials, thank you.";
				return;
			}
			
			if (txtLoginID.Text.Length == 0)
			{
				MessageBox.Show("You user Login ID is required.");
				return;
			}
			if (PasswordTextBox.Text.Length == 0)
			{
				MessageBox.Show("You user Login Password is required.");
				return;
			}
			if (txtCompanyID.Text.Length == 0)
			{
				MessageBox.Show("You user Company ID is required.");
				return;
			}
			if (cbRepoID.Text.Length == 0)
			{
				MessageBox.Show("Please select a repository.");
				return;
			}
			if (pwEncryptPW.Text.Length == 0)
			{
				MessageBox.Show("A Gateway password is required.");
				return;
			}
			
			if (modGlobals.gGateWayID < 0)
			{
				VerifySecureDbConnection();
				//MessageBox.Show("The Gateway has not been initiated, please attach to the Secure Gateway.")
				return;
			}
			
			lblAttachedMachineName.Text = modGlobals.gAttachedMachineName;
			lblCurrUserGuidID.Text = txtLoginID.Text;
			lblLocalIP.Text = modGlobals.gLocalMachineIP;
			lblServerInstanceName.Text = modGlobals.gServerInstanceName;
			lblServerMachineName.Text = modGlobals.gServerMachineName;
			lblCurrUserGuidID.Text = txtLoginID.Text;
			
			modGlobals.gCurrUserGuidID = txtLoginID.Text;
			modGlobals.gRepoID = cbRepoID.Text.Trim();
			
			//*************************
			SaveStaticVars();
			//*************************
			SB.Text = "Logging in, standby.";
			SB.Refresh();
			Application.DoEvents();
			
			string LoginID = txtLoginID.Text.Trim();
			string PW = PasswordTextBox.Text.Trim();
			string DecryptedPassword = "";
			string EncryptedPassword = "";
			
			bool DHX = DB.ckUserLoginExists(LoginID);
			int Tries = 1;
			if (DHX == true)
			{
				string tpw = DB.getBinaryPassword(LoginID);
				if (tpw.Trim().Length > 0)
				{
					if (tpw.ToUpper().Equals("PASSWORD"))
					{
						if (LoginID.ToUpper.Equals("ADMIN"))
						{
							string Msg = "";
							Msg = "It appears that this may be the first time into the system." + "\r\n";
							Msg += "As an ADMIN, you will be required to change your " + "\r\n" + "current password of \'password\' and," + "\r\n";
							Msg += "any users will have to be added to the system if they" + "\r\n" + "are allowed to access ECM Library." + "\r\n" + "\r\n";
							Msg += "Thank you for using ECM Library.";
							MessageBox.Show(Msg);
						}
PWOVER:
						frmPasswordChange.Default.ShowDialog();
						string NewPw = frmPasswordChange.Default.PW1;
						string NewPw2 = frmPasswordChange.Default.PW2;
						NewPw = NewPw.Trim;
						NewPw2 = NewPw2.Trim;
						if (NewPw.Equals(NewPw2))
						{
							string epw = ENC.AES256EncryptString(NewPw2);
							string sEnc = "Update users set UserPassword = \'" + epw + "\' where UserLoginID = \'" + LoginID + "\' ";
							bool BB2 = DB.ExecuteSqlNewConn(sEnc, false);
							if (BB2)
							{
								MessageBox.Show("Your password has been changed, for future logins, please use your new password.");
								PW = NewPw;
							}
							else
							{
								MessageBox.Show("Your password failed to update, please cntact an administrator or try to login again, closing down.");
								ProjectData.EndApp();
							}
						}
						else
						{
							Tries++;
							if (Tries >= 3)
							{
								MessageBox.Show("Too many tries, try logging in again from the start, closing down.");
								ProjectData.EndApp();
							}
							MessageBox.Show("The passwords do not equal each other, please reenter.");
							goto PWOVER;
						}
						
					}
				}
			}
			else
			{
				MessageBox.Show("Your user login ID has not been defined to the system, please contact an administrator, closing down.");
				ProjectData.EndApp();
			}
			
			frmPasswordChange.Default.Dispose();
			
			EncryptedPassword = ENC.AES256EncryptString(PW);
			
			string UPW = DB.getPw(LoginID);
			if (UPW.Equals(EncryptedPassword))
			{
				UID = LoginID;
				bGoodLogin = true;
				this.Hide();
				frmMain.Default.Show();
			}
			else
			{
				UID = "";
				bGoodLogin = false;
				MessageBox.Show("Failed to login, please verify your User ID and password.");
				this.Close();
			}
			
		}
		
		public void Cancel_Click(System.Object sender, System.EventArgs e)
		{
			modGlobals.bRunnner = false;
			this.Close();
		}
		
		
		public void Timer1_Tick(System.Object sender, System.EventArgs e)
		{
			//If HelpOn Then
			//    If bHelpLoaded Then
			//        TT.Active = True
			//    Else
			//        DB.getFormTooltips(Me, TT, True)
			//        TT.Active = True
			//        bHelpLoaded = True
			//    End If
			//Else
			//    TT.Active = False
			//End If
			//Application.DoEvents()
		}
		
		public void ckSaveAsDefaultLogin_CheckedChanged(System.Object sender, System.EventArgs e)
		{
			if (ckSaveAsDefaultLogin.Checked)
			{
				//Write the info (encrypted) out to a file for use during the next login.
				SavePersist();
			}
			else
			{
				//Remove the login file if it exists.
			}
		}
		
		public void PopulateRepoComboBox()
		{
			
			
			cbRepoID.Visible = false;
			pwEncryptPW.Visible = false;
			btnGetRepos.Visible = false;
			
			//MessageBox.Show("Get REPOS 01")
			
			string CS = "";
			string CompanyID = txtCompanyID.Text;
			bool RC = true;
			string RetMsg = "";
			
			if (CompanyID.Trim().Length == 0)
			{
				MessageBox.Show("Please supply your company ID and password.");
				return;
			}
			
			ProxySRCH.PopulateSecureLoginCB_V2(SecureID, strRepos, CompanyID, RC, RetMsg);
			
			string[] TempArray = strRepos.Split(Strings.Chr(254).ToString().ToCharArray());
			foreach (string s in TempArray)
			{
				if (s.Trim().Length > 0)
				{
					ListOfRepoIS.Add(s);
				}
			}
			
			cbRepoID.DataSource = ListOfRepoIS;
			
		}
		
		public void btnAttach_Click(System.Object sender, System.EventArgs e)
		{
			bool B = VerifySecureDbConnection();
			if (B)
			{
				OK.Visible = true;
				lblAttached.Text = "Attached";
			}
			else
			{
				OK.Visible = false;
				lblAttached.Text = "NOT Attached";
			}
			
		}
		
		public bool VerifySecureDbConnection()
		{
			
			string CompanyID = txtCompanyID.Text;
			string RepoID = cbRepoID.Text;
			bool RC = false;
			string RetMsg = "";
			
			
			if (CompanyID.Length == 0)
			{
				MessageBox.Show("CompanyID must be supplied, CANNOT ATTACH.");
				return false;
			}
			if (RepoID.Length == 0)
			{
				MessageBox.Show("A Repository must be selected, CANNOT ATTACH.");
				return false;
			}
			
			string EPW = pwEncryptPW.Text;
			EPW = ENC2.EncryptPhrase(EPW);
			
			ProxySRCH.validateAttachSecureLogin(SecureID, CompanyID, RepoID, txtLoginID.Text, EPW, RC, RetMsg);
			modGlobals.gCurrentConnectionString = (string) (ProxySRCH.getGatewayCS(SecureID, CompanyID, RepoID, EPW, RetMsg, RC));
			modGlobals.gCurrentConnectionString = ENC.AES256DecryptString(modGlobals.gCurrentConnectionString);
			modGlobals.gGateWayID = SecureID;
			
			return RC;
			
		}
		
		public void btnGetRepos_Click(System.Object sender, System.EventArgs e)
		{
			PopulateRepoComboBox();
			cbRepoID.Visible = true;
			pwEncryptPW.Visible = true;
			btnGetRepos.Visible = true;
			btnAttach.Visible = true;
		}
		
		public void SaveStaticVars()
		{
			
			if (SecureID >= 0)
			{
				modGlobals.gGateWayID = SecureID;
			}
			modGlobals.gRepoID = cbRepoID.Text;
			modGlobals.gCompanyID = txtCompanyID.Text;
			modGlobals.gUserID = txtLoginID.Text;
			modGlobals.gActiveGuid = CurrSessionGuid;
			modGlobals.gSessionGuid = CurrSessionGuid;
			modGlobals.gEncryptPW = pwEncryptPW.Text;
			
			//If Me.Resources.Contains("SecureID") Then
			//    Me.Resources.Remove("SecureID")
			//End If
			//Me.Resources.Add("SecureID", SecureID.ToString)
			//If Me.Resources.Contains("CurrSessionGuid") Then
			//    Me.Resources.Remove("CurrSessionGuid")
			//End If
			//Me.Resources.Add("CurrSessionGuid", CurrSessionGuid.ToString)
			//If Me.Resources.Contains("CompanyID") Then
			//    Me.Resources.Remove("CompanyID")
			//End If
			//Me.Resources.Add("CompanyID", txtCompanyID.Text)
			//If Me.Resources.Contains("RepoID") Then
			//    Me.Resources.Remove("RepoID")
			//End If
			//Me.Resources.Add("RepoID", cbRepoID.Text)
			//If Me.Resources.Contains("LoginID") Then
			//    Me.Resources.Remove("LoginID")
			//End If
			//Me.Resources.Add("LoginID", txtLoginID.Text)
			
			SavePersist();
			
			
		}
		
		public void SaveActiveParm(string ParmName, string ParmVal)
		{
			if (! bAttached)
			{
				return;
			}
			
			ProxySRCH.ActiveSession(SecureID, CurrSessionGuid, ParmName, ParmVal);
			
		}
		public void RemovePersist()
		{
			ISO.PersistDataInit("NA", "NA");
		}
		public void SavePersist()
		{
			ISO.PersistDataInit("CurrSessionGuid", CurrSessionGuid.ToString());
			ISO.PersistDataSave("CompanyID", txtCompanyID.Text);
			ISO.PersistDataSave("RepoID", cbRepoID.Text);
			ISO.PersistDataSave("LoginID", txtLoginID.Text);
			string TPW = pwEncryptPW.Text;
			TPW = ENC2.EncryptPhrase(TPW);
			ISO.PersistDataSave("EncryptPW", TPW);
			
			string UPW = PasswordTextBox.Text.Trim();
			UPW = ENC2.EncryptPhrase(UPW);
			ISO.PersistDataSave("UPW", UPW);
			
			ISO.PersistDataSave("EOD", "***");
		}
		
		public void getPersitData()
		{
			
			string SID = ISO.PersistDataRead("SecureID");
			if (SID.Length > 0)
			{
				SecureID = int.Parse(SID);
				modGlobals.gGateWayID = SecureID;
			}
			
			txtLoginID.Text = ISO.PersistDataRead("LoginID");
			
			txtCompanyID.Text = ISO.PersistDataRead("CompanyID");
			
			cbRepoID.Text = ISO.PersistDataRead("RepoID");
			if (cbRepoID.Text.Length > 0)
			{
				cbRepoID.SelectedItem = cbRepoID.Text;
			}
			
			string EPW = ISO.PersistDataRead("EncryptPW");
			pwEncryptPW.Text = ENC2.DecryptPhrase(EPW);
			
			string UPW = ISO.PersistDataRead("UPW");
			PasswordTextBox.Text = ENC2.DecryptPhrase(UPW);
			
			string sCurrSessionGuid = ISO.PersistDataRead("CurrSessionGuid");
			if (sCurrSessionGuid.Length > 0)
			{
				CurrSessionGuid = new Guid(sCurrSessionGuid);
			}
			else
			{
				CurrSessionGuid = new Guid();
			}
			
		}
		
		public void Timer2_Tick(System.Object sender, System.EventArgs e)
		{
			if (ckCancelAutoLogin.Checked == true)
			{
				Timer2.Enabled = false;
				
			}
			CurrLoginSecs++;
			int i = AutoLoginSecs - CurrLoginSecs;
			if (CurrLoginSecs == AutoLoginSecs)
			{
				lblMsg.Text = "Logging in...\"";
				OK_Click(null, null);
			}
			else
			{
				lblMsg.Text = (string) ("SECS to Login: " + i.ToString());
			}
			
		}
	}
	
}

File: Archiver/LoginForm1.vb
Imports System.Collections
Imports System.Collections.Generic

Imports System.IO
Imports ECMEncryption

Public Class LoginForm1

    Dim TRACEFLOW As String = System.Configuration.ConfigurationManager.AppSettings("TRACEFLOW")
    Dim CurrGatewaySvr As String = ""
    Dim AutoLoginSecs As Integer = 10
    Dim CurrLoginSecs As Integer = 0
    Dim LOG As New clsLogging
    Dim ISO As New clsIsolatedStorage
    Dim DBARCH As New clsDatabaseARCH
    Dim ENC As New ECMEncrypt
    Dim DMA As New clsDma
    Dim bHelpLoaded As Boolean = False
    Dim bAttached As Boolean = False
    Dim bPopulateComboBusy As Boolean = False
    Dim AutoSecs As Integer = 1
    Dim Tries As Integer = 1

    Public bGoodLogin As Boolean = False
    Public UID As String = ""

    Dim CurrSessionGuid As Guid = Guid.NewGuid

    Dim SecureID As Integer = -1

    Dim ListOfRepoIS As New System.Collections.Generic.List(Of String)
    Dim strRepos As String = ""
    Private osInfo As OperatingSystem

    Dim FQN As String = ""
    Dim CurrUserName As String = ""
    Dim dDebug As String = System.Configuration.ConfigurationManager.AppSettings("debug_frmReconMain")

    Private Sub LoginForm1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load


        If TRACEFLOW = 1 Then DBARCH.RemoteTrace(9900, "LoginIn001", "00")

        lblExecTime.Visible = False
        btnStopExec.Visible = False

        Me.TopMost = False

        'InitPersistData()

        If TRACEFLOW.Equals(1) Then
            LOG.WriteToTraceLog("**********************************************************************")
            LOG.WriteToTraceLog("Start of Trace for LoginForm" + Now.ToString)
        End If

        Dim OSVersion As String = Environment.OSVersion.ToString()
        Dim OSVersionMajor As String = Environment.OSVersion.Version.Major.ToString
        Dim OSVersionMinor As String = Environment.OSVersion.Version.Minor.ToString
        Dim OSVersionMinorRev As String = Environment.OSVersion.Version.MinorRevision.ToString

        AddTemDir()

        If dDebug.Equals("1") Then
            LOG.WriteToTraceLog(Now.ToString + "Step 01 :Added the temp directory")
        End If

        If TRACEFLOW = 1 Then DBARCH.RemoteTrace(9900, "LoginIn001", "01")


        If TRACEFLOW = 1 Then DBARCH.RemoteTrace(9900, "LoginIn002", "02")
        CurrUserName = Environment.UserName

        If TRACEFLOW = 1 Then DBARCH.RemoteTrace(9900, "LoginIn003", "03")

        getPersitData()
        If TRACEFLOW = 1 Then DBARCH.RemoteTrace(9900, "LoginIn004", "04")

        lblAttachedMachineName.Text = LOG.getEnvVarMachineName
        lblNetworkID.Text = LOG.getEnvVarNetworkID

        If TRACEFLOW = 1 Then DBARCH.RemoteTrace(9900, "LoginIn005", "05")
        bRunnner = False

        If (UseDirectoryListener.Equals(0)) Then
            ckDisableListener.Visible = False
        End If

        'DBARCH.ckUpdateTbl()

    End Sub

    Private Sub OK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OK.Click

        Timer2.Enabled = False
        Timer3.Enabled = False

        If PasswordTextBox.Text.ToUpper.Equals("PASSWORD") Then
            If txtLoginID.Text.Length = 0 Then
                MessageBox.Show("Please enter your USERID into the Login ID field.")
                Return
            End If
            frmPasswordChange.ShowDialog()
            SB.Text = "Please login using your new credentials, thank you."
            Return
        End If

        If txtLoginID.Text.Length = 0 Then
            MessageBox.Show("Your user Login ID is required.")
            Return
        End If
        If PasswordTextBox.Text.Length = 0 Then
            MessageBox.Show("Your user Login Password is required.")
            Return
        End If

        lblAttachedMachineName.Text = gAttachedMachineName
        lblCurrUserGuidID.Text = txtLoginID.Text
        lblLocalIP.Text = gLocalMachineIP
        lblServerInstanceName.Text = gServerInstanceName
        lblServerMachineName.Text = gServerMachineName
        lblCurrUserGuidID.Text = txtLoginID.Text

        gCurrUserGuidID = txtLoginID.Text

        '*************************
        SaveStaticVars()
        '*************************
        SB.Text = "Logging in, standby."
        SB.Refresh()
        Application.DoEvents()

        Dim LoginID As String = txtLoginID.Text.Trim
        Dim PW As String = PasswordTextBox.Text.Trim
        Dim DecryptedPassword As String = ""
        Dim EncryptedPassword As String = ""
        Dim RetCode As Boolean = False

        Dim DHX As Integer = 99
        Dim ix As Integer = 0
        Dim rMsg As String = ""

        DHX = DBARCH.ckUserLoginExists(LoginID, RetCode, rMsg)
        ix = DHX

        If ix < 1 And rMsg.Length > 0 Then
            MessageBox.Show("FAILED TO ATTACH USER: " + rMsg)
        ElseIf ix < 1 Then
            MessageBox.Show("ERROR 01Q: FAILED TO ATTACH USER - verify your ID and password ")
        End If

        If ix = 1 Then
            Dim tpw As String = DBARCH.getBinaryPassword(LoginID)
            'MessageBox.Show("REMOVE THIS: TPW=" + tpw.ToString)
            If tpw.Trim.Length > 0 Then
                If tpw.ToUpper.Equals("PASSWORD") Then
                    If LoginID.ToUpper.Equals("ADMIN") Then
                        Dim Msg As String = ""
                        Msg = "It appears that this may be the first time into the system." + Environment.NewLine
                        Msg += "As an ADMIN, you will be required to change your " + Environment.NewLine + "current password of 'password' and," + Environment.NewLine
                        Msg += "any users will have to be added to the system if they" + Environment.NewLine + "are allowed to access ECM Library." + Environment.NewLine + Environment.NewLine
                        Msg += "Thank you for using ECM Library."
                        MessageBox.Show(Msg)
                    End If
PWOVER:
                    frmPasswordChange.ShowDialog()
                    Dim NewPw As String = frmPasswordChange.PW1
                    Dim NewPw2 As String = frmPasswordChange.PW2
                    NewPw = NewPw.Trim
                    NewPw2 = NewPw2.Trim
                    If NewPw.Equals(NewPw2) Then
                        Dim epw As String = ENC.AES256EncryptString(NewPw2)
                        Dim sEnc As String = "Update users set UserPassword = '" + epw + "' where UserLoginID = '" + LoginID + "' "
                        Dim BB2 As Boolean = DBARCH.ExecuteSqlNewConn(sEnc, False)
                        If BB2 Then
                            MessageBox.Show("Your password has been changed, for future logins, please use your new password.")
                            PW = NewPw
                        Else
                            MessageBox.Show("Your password failed to update, please cntact an administrator or try to login again, closing down.")
                            End
                        End If
                    Else
                        Tries += 1
                        If Tries >= 3 Then
                            MessageBox.Show("Too many tries, try logging in again from the start, closing down.")
                            End
                        End If
                        MessageBox.Show("The passwords do not equal each other, please reenter.")
                        GoTo PWOVER
                    End If
                End If
            End If
        Else
            MessageBox.Show("Your user login ID has not been defined to the system, please contact an administrator, closing down.")
            GoTo SKIPOUT
        End If

        frmPasswordChange.Dispose()

        EncryptedPassword = ENC.AES256EncryptString(PW)

        'MessageBox.Show("Remove this later WDM")

        Dim UPW As String = DBARCH.getPw(LoginID)
        If UPW.Equals(EncryptedPassword) Then
            UID = LoginID
            bGoodLogin = True
            Timer2.Enabled = False

            Me.Hide()

            Dim MainPage As frmMain = New frmMain
            MainPage.Show()

            Return
        Else
            UID = ""
            bGoodLogin = False
            MessageBox.Show("ERROR A12: Failed to login, please verify your User ID and password! ")
            Tries += 1
            If Tries >= 3 Then
                MessageBox.Show("Too many tries, try logging in again from the start, closing down.")
                End
            End If
        End If
SKIPOUT:
    End Sub

    Private Sub Cancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cancel.Click
        bRunnner = False
        Me.Close()
    End Sub

    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        'If HelpOn Then
        '    If bHelpLoaded Then
        '        TT.Active = True
        '    Else
        '        DBARCH.getFormTooltips(Me, TT, True)
        '        TT.Active = True
        '        bHelpLoaded = True
        '    End If
        'Else
        '    TT.Active = False
        'End If
        'Application.DoEvents()
    End Sub

    Private Sub ckSaveAsDefaultLogin_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ckSaveAsDefaultLogin.CheckedChanged
        If ckSaveAsDefaultLogin.Checked Then
            'Write the info (encrypted) out to a file for use during the next login.
            SavePersist()
        Else
            'Remove the login file if it exists.
        End If
    End Sub

    Sub SaveStaticVars()

        If SecureID >= 0 Then
            gGateWayID = SecureID
            gSecureID = SecureID
        End If

        gUserID = txtLoginID.Text
        gActiveGuid = CurrSessionGuid
        gSessionGuid = CurrSessionGuid

        SavePersist()

    End Sub

    ''' <summary>
    ''' Saves the active parm.
    ''' </summary>
    ''' <param name="ParmName">Name of the parm.</param>
    ''' <param name="ParmVal"> The parm value.</param>
    'Sub SaveActiveParm(ByVal ParmName As String, ByVal ParmVal As String)
    '    If Not bAttached Then
    '        Return
    '    End If

    '    ProxySearch.ActiveSession(SecureID, CurrSessionGuid, ParmName, ParmVal)

    'End Sub

    Sub InitPersistData()
        ISO.PersistDataInit("CurrSessionGuid", "")
        ISO.PersistDataSave("CompanyID", "")
        ISO.PersistDataSave("RepoID", "")
        ISO.PersistDataSave("LoginID", "")
        ISO.PersistDataSave("EncryptPW", "")
        ISO.PersistDataSave("UPW", "")
        ISO.PersistDataSave("EOD", "***")
    End Sub

    Sub RemovePersist()
        ISO.PersistDataInit("NA", "NA")
    End Sub

    Sub SavePersist()
        'EncryptPhrase
        ISO.PersistDataInit("CurrSessionGuid", CurrSessionGuid.ToString)
        ISO.PersistDataSave("LoginID", txtLoginID.Text)
        TPW = ENC.AES256EncryptString(PasswordTextBox.Text.Trim)
        ISO.PersistDataSave("EncryptPW", TPW)

        ISO.PersistDataSave("AutoExecute", ckAutoExecute.Checked.ToString)

        Dim UPW As String = PasswordTextBox.Text.Trim
        UPW = ENC.AES256EncryptString(UPW)
        ISO.PersistDataSave("UPW", UPW)

        ISO.PersistDataSave("EOD", "***")
    End Sub

    Sub getPersitData()

        Dim SID As String = ISO.PersistDataRead("SecureID")
        If SID.Length > 0 Then
            SecureID = CInt(SID)
            gSecureID = SecureID
            gGateWayID = SecureID
        End If

        Dim strAutoExec As String = ISO.PersistDataRead("AutoExecute")
        If strAutoExec.ToUpper.Equals("TRUE") Then
            gAutoExec = True
            ckAutoExecute.Checked = True
            ckAutoExecute.CheckState = 1
            Timer3.Enabled = True
            btnStopExec.Visible = True
            lblExecTime.Visible = True
        Else
            gAutoExec = False
            ckAutoExecute.Checked = False
            ckAutoExecute.CheckState = 0
            Timer3.Enabled = False
            btnStopExec.Visible = False
            lblExecTime.Visible = False
        End If

        txtLoginID.Text = ISO.PersistDataRead("LoginID")

        Dim UPW As String = ISO.PersistDataRead("UPW")
        PasswordTextBox.Text = ENC.AES256DecryptString(UPW)

        Dim sCurrSessionGuid As String = ISO.PersistDataRead("CurrSessionGuid")
        If sCurrSessionGuid.Length > 0 Then
            CurrSessionGuid = New Guid(sCurrSessionGuid)
        Else
            CurrSessionGuid = New Guid()
        End If

    End Sub

    Private Sub Timer2_Tick(sender As System.Object, e As System.EventArgs) Handles Timer2.Tick
        If (ckCancelAutoLogin.Checked = True) Then
            Timer2.Enabled = False
        End If
        CurrLoginSecs += 1
        Dim i As Integer = AutoLoginSecs - CurrLoginSecs
        If CurrLoginSecs = AutoLoginSecs Then
            lblMsg.Text = "Logging in..."""
            OK_Click(Nothing, Nothing)
        Else
            lblMsg.Text = "SECS to Login: " + i.ToString
        End If

    End Sub

    Private Sub AddTemDir()
        If Not Directory.Exists("c:\TempUploads") Then
            Directory.CreateDirectory("c:\TempUploads")
        End If
    End Sub

    Private Sub cbRepoID_SelectedIndexChanged(sender As Object, e As EventArgs)
        'ISO.PersistDataSave("CompanyID", txtCompanyID.Text)
        'ISO.PersistDataSave("RepoID", cbRepoID.Text)

        'gCustomerID = txtCompanyID.Text
        'gRepoID = cbRepoID.Text

        'setGatewayEndpoints()
        Timer2.Enabled = False
    End Sub

    Private Sub txtCompanyID_TextChanged(sender As Object, e As EventArgs)
        Timer2.Enabled = False
    End Sub

    Private Sub ckCancelAutoLogin_CheckedChanged(sender As Object, e As EventArgs) Handles ckCancelAutoLogin.CheckedChanged

    End Sub

    Private Sub PictureBox1_MouseEnter(sender As Object, e As EventArgs) Handles PictureBox1.MouseEnter
        SB.Text = "D. Miller & Asso., LTD., Chicago, IL."
    End Sub

    Private Sub PictureBox1_MouseLeave(sender As Object, e As EventArgs) Handles PictureBox1.MouseLeave
        SB.Text = ""
    End Sub

    Private Sub btnChgPW_Click(sender As Object, e As EventArgs) Handles btnChgPW.Click

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim LoginID As String = txtLoginID.Text.Trim
        Dim UPW As String = DBARCH.getPw(LoginID)
        Dim EncPassword = ENC.AES256EncryptString(UPW)
        If UPW.Equals(EncPassword) Then
            MessageBox.Show("Password is good")
        Else
            MessageBox.Show("Password is BAD: " + UPW + " : " + EncPassword)
        End If
    End Sub

    Private Sub ckAutoExecute_CheckedChanged(sender As Object, e As EventArgs) Handles ckAutoExecute.CheckedChanged
        If ckAutoExecute.Checked And Not ckSaveAsDefaultLogin.Checked Then
            If gAutoExec.Equals(False) Then
                MessageBox.Show("Auto Execute cannot be set until Save As Default Login is set and checked... returning.")
            End If
        End If
        If ckAutoExecute.Checked And ckSaveAsDefaultLogin.Checked Then
            SavePersist()
        End If


    End Sub

    Private Sub btnStopExec_Click(sender As Object, e As EventArgs) Handles btnStopExec.Click
        Timer3.Enabled = False
        SB.Text = "Auto Execute Cancelled"
        gAutoExec = False
        btnStopExec.Visible = False
        lblExecTime.Visible = False
    End Sub

    Private Sub Timer3_Tick(sender As Object, e As EventArgs) Handles Timer3.Tick
        AutoSecs += 1
        lblExecTime.Text = "Executing in " + (10 - AutoSecs).ToString + " Secs."
        lblExecTime.Refresh()
        If AutoSecs.Equals(10) Then
            Console.WriteLine("Auto-execution started")
            OK.PerformClick()
        End If
    End Sub

    Private Sub ckDisableListener_CheckedChanged(sender As Object, e As EventArgs) Handles ckDisableListener.CheckedChanged
        If ckDisableListener.Checked Then
            gTempDisableDirListener = True
        Else
            gTempDisableDirListener = False
        End If
    End Sub
End Class